RC,Coversation pages and Navbar wired with props

harsh010102 opened the pull request:
1.Added an additional list interface for ResearchContextPage and ConversationPage.
`interface ConversationPageProps {   convs: Conversation[]; }`

2.edited story files for both pages to render an randomly generated list

`export const WithManyConversations: Story = { args: { convs: [ { id: 3948887085, title: "AzOc5jIYjLyCNnZU9KzUMg", }, ] }`

3.Added {activelink} prop to Navbar, also fixed the highlight-when-hover problem.

4. aligned the "ResearchContext | Coversations | About Us ".

NOTE: the darkmode issue and modal-open issue still persist.

alebg commented on tests/components/researchcontextpage.test.tsx:
```diff
@@ -0,0 +1,74 @@
+// import { expect, describe, it } from "vitest";
```
Please clean this

alebg commented on tests/components/conversationpage.test.tsx:
```diff
@@ -0,0 +1,78 @@
+// import { expect, describe, it } from "vitest";
```
And clean this as well

alebg commented:
Great! Thanks, this is looking much better and I can actually test. A few remarks:

1. In the NavBar storybook, when I change the 'activePage' value I don't see any changes in the bar, only when I hover over the words. You could make it a harcoded selection between the three options in the storybook, instead of having to write a string. Ask @iamMihirT how he did it for the 'user' and 'llm' options in the Chatpage

2. Please use props to inject the api URLs for the calls. For example, you could add another attribute to your ResearchContextPageProps called "apiUrl" (string), and then use it with `axios.get(apiUrl)`

3. Please clean the comments you left in the tests

4. I like the design and the conversations and RCs going horizontally (to potentially visualize better long titles/descriptions), but let me validate with Manny if this design is OK and I'll come back to you

alebg commented:
@harsh010102 some additional remarks:

4. The design looks fine, I just validated it with manny.

5. On ResearchContextPage, please add a "New Reseach Context Button" (just like you did for COnversations), which this time will accept 'title' and 'description'.
For both buttons, please use an "OnClick" from which we can pass down values from the actual API calls. Please check this example: https://github.com/dream-aim-deliver/ralph-the-moose-ui-kit/blob/main/lib/components/button/Button.tsx
The "OnClick" method will be injected from the outside, so leave it as in the example.

Please also do more robust tests for this:
https://github.com/dream-aim-deliver/ralph-the-moose-ui-kit/blob/main/tests/components/Button.test.tsx

harsh010102 commented:
1. Navbar updated , referred @iamMihirT code and added
`  role?: "Research Context" | "Conversations" | "About Us";
` 
2. added prop apiUrl in both pages(RC and Conv)
`interface ConversationPageProps {
  convs: Conversation[];
  apiUrl: string;
}`
3. Added "+New Research Context" button, modal with 2 text fields (title: and description: )

alebg commented:
@harsh010102 great job! Looking much better, app builds and test pass. Some remarks on a visual test:

1a. The card components in the conversation page break when the title has long strings without spaces
1b. The prop named "id" gets rendered as if it were a "description". You should disable this and render only the title, as we won't have any more information about a conversation
1c. As a design, maybe wrapping the title of the conversation instead of cutting it with "..." is better, and let the card expand a bit more horizontally, so people can read their conversation titles instead of hiding it. But visually test yourself first to see if that makes sense/doesn't look to ugly

![imagen](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/62880920/8365494e-1ca5-483f-a805-4f8a012bfc9b)

2. Exactly the same three remarks (1a, 1b, 1c) for the Research Context page:

![imagen](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/62880920/944cd9c7-f07d-4538-bd15-7e62b4f76548)

3. The props of the NavBar don't really do anything:

![imagen](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/62880920/89c08b0f-6287-4fee-8cd4-ad69d9d3494c)

harsh010102 commented:
![navbar-working](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/72195967/ac11ebc9-f4a6-4a22-9c8b-a32d002d506c)
hii Luis , the navbar is working fine on my local machine.

harsh010102 commented:
![image](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/72195967/fa61e545-09ad-4ad6-9578-bfddf60fa405)

check this out @alebg there is a tailwind class "text-balance" it can be use for text-wrapping but unfortunately when I pass a very very long single string it overflows. but if there is a long spaced-out title it will be wrapped inside the card.

I have made some minor changes to font size and padding, also increased the size of card. let me know how it looks then i will make appropriate changes and push.

alebg commented:
Hi @harsh010102 !

1. It'd be best if the card is not visually breaking at all and the behaviour is consistent, under any condition, including having a very long string. If tailwind is not doing the job you could try to find another way

2. I think it's my browser's dark mode messing up with the component. I just changed to light mode (not in storybook, but browser-wise) and the prop does work. But this is not good enough. Some real users will be having browsers in dark mode like me and for them the component is not going to work. You should investigate on why this is happening, together with @iamMihirT if possible. Try to replicate by using firefox or chrome in dark mode. For me in both this issue happens, and I reiterate that I cannot really read the institute's name. We had talked about that already but I hadn't had time to fix that, so if you could find why this is happening and a solution, that would be great:

![image](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/62880920/ab91bc7f-8337-4101-a07c-cf43fdfeda0f)

If this was happening in our production app, you wouldn't tell end users "it works on fine on my local machine", right? ;)
An idea for a hacky solution is just hard code the same background color and letter color for everyone under any condition. But it'd be nice to have a proper dark mode that works and that allows me to read the institute name and see the component working as it should

harsh010102 commented:
1. Removed darkmode toggle from Navbar, instead use Tailwind class `dark:bg-slate-800` to enable darkmode based on user's environment.

alebg commented:
Hi @harsh010102 . Reviewing your latest commits, I think everything is fine now, except for the dark mode problem (for which I think you're sending another PR). One small detail, research contexts will have descriptions and they should render below the title. What we're rendering now is their IDs, but this doesn't make much sense:

![imagen](https://github.com/dream-aim-deliver/planckster-ui-kit/assets/62880920/af8ae59b-4de0-4360-8bfa-ba963a68bc84)

I.e., research contexts will have three attributes, ID, title, and description, and we only want to render title and description (description should go where ID is now)

alebg commented on lib/components/researchcontextpage/researchcontextpage.tsx:
```diff
@@ -0,0 +1,181 @@
+// import React, { useState, useEffect } from "react";
```
@harsh010102 please delete everything that's not being used

alebg commented:
@harsh010102 Alright, looks good, look just at my latest comment on the comments you left that should be deleted. After that, I'll merge
